### ISSUE TITLE
Negative ranking on leaderboards _only_ when accessing the global leaderboards from the lowest ranked player's userpage

### ISSUE THREAD

**Issue by Flyspeck101** [tetrio/issues#998]
### Checklist

- [X] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this issue is not already reported.
- [X] I am using the **newest version** of TETR.IO.
- [X] I am using a **supported browser** or the Desktop Client.
- [X] I am not using anything that **modifies game code** (e.g. TETR.IO PLUS).
- [X] I am certain this is a **bug**, not a feature request.

### OS and Version

Windows 7 

### Browser and version

Chrome 104 

### Describe the bug/issue

In the TETRA CHANNEL leaderboard, when viewing the lowest ranked player, the ranks are negative. The lowest ranked person is #0, and the second lowest is #-1, so on. 

### Reproducing

1. Go to the lowest ranked person's TETRA CHANNEL profile (currently DIETZNUTS)
2. Click on the Global Rank button, which takes you to the leaderboard, which now has negative values. 

### Expected behavior

The leaderboard should show positive rankings, in this case it should be #42469. 

### Screenshots

![image](https://user-images.githubusercontent.com/74536088/184479877-bc043146-e24d-4203-8166-80a3234049cb.png)

### Console logs

Nothing out of the ordinary. "Failed to load resource" should have no relation to this bug. 
![image](https://user-images.githubusercontent.com/74536088/184480126-c33bbb15-22e4-47d0-be71-7d5db508a69b.png)

**Comment by ZaptorZap** [comment]
Nah, still happens.

**Comment by Flyspeck101** [comment]
Just checked, it also occurs on an individual country's lowest ranked player if you click the country leaderboard from their Tetra Channel profile.

**Comment by swng** [comment]
Stepping through the [players.js](https://ch.tetr.io/res/js/players.js) code:

(L37-L47) Tetra Channel grabs a "waypoint" position and TR value. In the bugged case, this lowest ranked player's position and TR. These values are stored in object `contextPos`. It then calls the Load() function with these values (L208)

(L208-L222) Tetra Channel then makes its first API call with an `after` query, grabbing all players underneath. In this case, no one.

(L228) Tetra Channel then populates the page with all players underneath. In this case, no one.

(L265) "also load above us". Tetra Channel loads the players above the waypoint position. It calls function GetTopWaypointPosition() to determine the rank of the top player on the page. This is so the players above can be ranked appropriately.

(L144) GetTopWaypointPosition() reads the DOM to grab `data-position` of the top player currently on the page. In this case, there is no one on the page. So instead, the function returns 1.

The page is now populated with the players ranked better, with a for loop starting with this TopWaypointPosition. Worst ranked player is assigned 0th place, then the next player -1 then -2 then -3 as per the for loop.

As you can see, the page attempted to determine TopWaypointPosition by reading the page, but the page was empty, so it returned the default value of 1, and this causes the negative rankings.

Is it necessary to read the page to determine TopWaypointPosition? The correct value in this case (`firstLoad`) is already known and stored in `contextPos.position` (+1).

TL;DR

I believe this issue can be fixed by adjusting L266 from
`			Load(true, GetTopWaypointTR(), GetTopWaypointPosition(), country, false, true);
`
to
`			Load(true, GetTopWaypointTR(), contextPos.position + 1, country, false, true);
`

**Comment by o5k** [comment]
Hotfixed - F5 to receive the fix. Thank you swng for the help—the getters are indeed not logical to be used on first load.

**Comment by ZaptorZap** [comment]
Not gonna begin to recommend another fix, but it's clear to me that this still has some improving to do. Previously, this wouldn't happen, but now, it appears as though when following the country placement link from a userpage associated with a low playercount country, the below anomaly occurs:
![image](https://github.com/tetrio/issues/assets/64891705/e8927954-11f3-4347-9f73-0be0241f62c6)

I believe this only began happening after this hotfix. I only recently found out about the hotfix, hence the delay.

**Comment by swng** [comment]
I see the issue. We used as a fix for L266:
`			Load(true, waypointTR-0.00001, waypointPosition+1, country, false, true);
`

However, I didn't realize that we're mutating waypointPosition above this call...

L228 - 230, a for loop including a `++waypointPosition`

So, uh, we could use a different variable: Assign some `let waypointPosition_copy = waypointPosition` at the start of Load(), and use `Load(true, waypointTR-0.00001, waypointPosition_copy+1, country, false, true);` on L266 instead.

Sorry for the oversight.

**Comment by ZaptorZap** [comment]
Seems like the fix was implemented in [6.4.2](https://tetr.io/about/patchnotes/#chlog_6_4_2).
